journal/src/main/scala/com/evolutiongaming/kafka/journal/eventual/EventualRead.scala

```diff
@@ -0,0 +1,45 @@
+package com.evolutiongaming.kafka.journal.eventual
+
+import cats.~>
+import cats.implicits._
+import com.evolutiongaming.catshelper.MonadThrowable
+import com.evolutiongaming.kafka.journal.{JournalError, JsonCodec, Payload, PayloadType}
+
+import scala.util.Try
+
+trait EventualRead[F[_], A] {
+
+  def readEventual(payloadAndType: EventualPayloadAndType): F[A]
```

Comment:
why not apply ?

Reply:
named them so in order to have readKafka and readEventual in JournalRead trait (which extends EventualRead and KafkaRead), instead of having two apply methods.
can rename them to apply and stick with overloaded methods though, if you think that's better

Reply:
maybe having single instance extending both is not that good idea? :)

Reply:
yeah.. didn't want to pass two things around, but maybe that's okay..
will split and rename then